goal_src/jak2/levels/city/common/vehicle-control.gc

```diff
@@ -147,30 +147,34 @@
     (let* ((s5-0 (-> obj branch dest-node))
            (s4-0 (-> s5-0 branch-count))
            )
-      (cond
-        ((= s4-0 1)
-         (let ((v1-4 (-> s5-0 branch-array 0)))
-           (if (!= (-> v1-4 dest-node id) #xffff)
-               (set! gp-0 v1-4)
-               )
-           )
-         )
-        ((< 1 s4-0)
-         (let ((v1-6 (rand-vu-int-count s4-0)))
-           (countdown (a0-8 s4-0)
-             (let ((a1-3 (-> s5-0 branch-array v1-6)))
-               (b! (not (!= (-> a1-3 dest-node id) #xffff)) cfg-8 :delay (empty-form))
-               (set! gp-0 a1-3)
-               )
-             (b! #t cfg-12 :delay (nop!))
-             (label cfg-8)
-             (+! v1-6 1)
-             (if (>= v1-6 s4-0)
-                 (set! v1-6 0)
-                 )
-             )
-           )
-         )
+      (b! (!= s4-0 1) cfg-4 :delay (empty-form))
+      (let ((v1-4 (-> s5-0 branch-array 0)))
+        (if (!= (-> v1-4 dest-node id) #xffff)
+            (set! gp-0 v1-4)
+            )
+        )
+      (b! #t cfg-12 :delay (nop!))
+      (label cfg-4)
+      (when (< 1 s4-0)
+        (let ((v1-6 (rand-vu-int-count s4-0))
+              (a0-8 s4-0)
+              )
+          (b! #t cfg-10 :delay (nop!))
+          (label cfg-6)
+          (+! a0-8 -1)
+          (let ((a1-3 (-> s5-0 branch-array v1-6)))
+            (b! (not (!= (-> a1-3 dest-node id) #xffff)) cfg-8 :delay (empty-form))
+            (set! gp-0 a1-3)
+            )
+          (b! #t cfg-12 :delay (nop!))
+          (label cfg-8)
+          (+! v1-6 1)
+          (if (>= v1-6 s4-0)
+              (set! v1-6 0)
+              )
+          (label cfg-10)
+          (b! (nonzero? a0-8) cfg-6 :delay (nop!))
+          )
```

Comment:
weird, this looks worse now

Reply:
Yeah I don't really have any idea what happened in this file.  I thought I had it completely decompiled, but then found out 2 methods didn't have their CFGs fixed (this one included).  So I'm still totally confused what happened lol